controllers/trustyaiservice_controller.go

```diff
@@ -138,10 +143,15 @@ func (r *TrustyAIServiceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	instance.Status.Ready = corev1.ConditionTrue
 
 	// CR found, add or update the URL
-	err = updatePayloadProcessor(ctx, r.Client, "mlserver", payloadProcessorName, instance.Name, instance.Namespace, false)
+	// Call the function to patch environment variables for Deployments that match the label
+	shouldContinue, err := r.patchEnvVarsByLabelForDeployments(ctx, req.Namespace, modelMeshLabelKey, modelMeshLabelValue, payloadProcessorName, req.Name, false)
 	if err != nil {
-		log.FromContext(ctx).Error(err, "Failed to update ModelMesh payload processor.")
-		// handle error
+		log.FromContext(ctx).Error(err, "Could not patch environment variables for Deployments.")
+		return ctrl.Result{}, err
+	}
+	if !shouldContinue {
+		log.FromContext(ctx).Info("Not all replicas are ready, requeue the reconcile request")
```

Comment:
This is more a side comment but why do we need to wait all replicas to be up before doing the injection of the variable? Is there any undesired side effect or it _just_ don't work? :)

Reply:
It will work, this is the previous logic (no longer needed). It will be removed with
- #57